### ISSUE TITLE
Redirect unregistered routes to their localized version

### ISSUE THREAD

**Issue by ivanvermeyen** [codezero-be/laravel-localized-routes#18]
It seems to be a common practice to redirect any unregistered URL that does not have a supported locale to its localized version:

Accessing `/dashboard` would redirect to `/en/dashboard` if `en` is the detected locale. If `/en/dashboard` is a registered route, it would just show that page.

Accessing `/non/existing/route` would redirect to `/en/non/existing/route` if `en` is the detected locale, which would then show a 404 page in the appropriate language.

Accessing `/nl/non/existing/route` would show the 404 page in Dutch, if `nl` is a supported locale.

I will need to research how to implement this with and/or without using fallback routes.

**Comment by Jatarii** [comment]
Do you have already a working exemple? This would be really helpful!

**Comment by ivanvermeyen** [comment]
Hi, unfortunately I have some other projects to prioritize at the moment.
I will be implementing new features once I have the time to spare. 👍

**Comment by Jatarii** [comment]
Would love to see this coming! Thanks!

**Comment by pascalbaljet** [comment]
For anyone looking for a solution, I'm currently using the `fallback` method to redirect unlocalized routes:

```php
use CodeZero\LocalizedRoutes\LocalizedUrlGenerator;
use CodeZero\Localizer\Middleware\SetLocale;
use Illuminate\Support\Facades\Route;
use Illuminate\Support\Facades\URL;

Route::fallback(function ($path = null) {
    $localizedUrl = app(LocalizedUrlGenerator::class)->generateFromRequest();

    if ($localizedUrl !== URL::to($path)) {
        return redirect()->to($localizedUrl);
    }

    abort(404);
})->middleware(SetLocale::class);
```

**Comment by ivanvermeyen** [comment]
Tagged and released in 2.5.0 🎉
